testing/scenario_app/bin/run_android_tests.dart

```diff
@@ -7,188 +7,80 @@ import 'dart:convert';
-          ? join(
-              engine.srcDir.path,
-              'third_party',
-              'android_tools',
-              'sdk',
-              'platform-tools',
-              'adb',
-            )
-          : null,
-    )
-    ..addOption(
-      'ndk-stack',
-      help: 'Path to the ndk-stack tool',
-      defaultsTo: engine != null
-          ? join(
-              engine.srcDir.path,
-              'third_party',
-              'android_tools',
-              'ndk',
-              'prebuilt',
-              () {
-                if (Platform.isLinux) {
-                  return 'linux-x86_64';
-                } else if (Platform.isMacOS) {
-                  return 'darwin-x86_64';
-                } else if (Platform.isWindows) {
-                  return 'windows-x86_64';
-                } else {
-                  throw UnsupportedError('Unsupported platform: ${Platform.operatingSystem}');
-                }
-              }(),
-              'bin',
-              'ndk-stack',
-            )
-          : null,
-    )
-    ..addOption(
-      'out-dir',
-      help: 'Out directory',
-      defaultsTo: engine
-          ?.outputs()
-          .where((Output o) => basename(o.path.path).startsWith('android_'))
-          .firstOrNull
-          ?.path
-          .path,
-    )
-    ..addOption(
-      'smoke-test',
-      help: 'Runs a single test to verify the setup',
-    )
-    ..addFlag(
-      'use-skia-gold',
-      help: 'Use Skia Gold to compare screenshots.',
-      defaultsTo: isLuciEnv,
-    )
-    ..addFlag(
-      'enable-impeller',
-      help: 'Enable Impeller for the Android app.',
-    )
-    ..addOption(
-      'output-contents-golden',
-      help: 'Path to a file that contains the expected filenames of golden files.',
-      defaultsTo: engine != null
-          ? join(
-              engine.srcDir.path,
-              'flutter',
-              'testing',
-              'scenario_app',
-              'android',
-              'expected_golden_output.txt',
-            )
-          : null,
-    )
-    ..addOption(
-      'impeller-backend',
-      help: 'The Impeller backend to use for the Android app.',
-      allowed: <String>['vulkan', 'opengles'],
-      defaultsTo: 'vulkan',
-    )
-    ..addOption(
-      'logs-dir',
-      help: 'The directory to store the logs and screenshots. Defaults to '
-          'the value of the FLUTTER_LOGS_DIR environment variable, if set, '
-          'otherwise it defaults to a path within out-dir.',
-      defaultsTo: Platform.environment['FLUTTER_LOGS_DIR'],
+  // Parse the command line arguments.
+  final Options options;
+  try {
+    options = Options.parse(
+      args,
+      environment: environment,
+      localEngine: localEngineDir,
     );
+  } on FormatException catch (error) {
+    stderr.writeln(error);
+    stderr.writeln(Options.usage(
+      environment: environment,
+      localEngineDir: localEngineDir,
+    ));
+    exitCode = 1;
+    return;
```

Comment:
I found that uncaught exceptions will also set the exit code.  Can we rely on that instead of setting the exitCode and trying to short circuit the logic?

Reply:
It's a caught exception though.

I could `rethrow` but then the output will look meh?

Reply:
It's ugly but informative for developers since it will have the stack trace.  I ended up going the uncaught route since trying to manually short circuit and maintain a global variable (exitCode) seemed to be fighting the language.  You're call.  I may even have ended up doing it both ways in different places =P